src/contracts/utils/ERC1155MetadataPrefix.sol

```diff
@@ -0,0 +1,72 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.7.4;
+
+import "../interfaces/IERC1155Metadata.sol";
+import "./Ownable.sol";
+
+
+contract ERC1155MetadataPrefix is IERC1155Metadata, Ownable {
+  string public uriPrefix;
+
+  event URIPrefixChanged(string _uriPrefix);
+
+  bool immutable includeAddress;
+
+  constructor(string memory _prefix, bool _includeAddress) Ownable(msg.sender) {
```

Comment:
msg.sender will always be the factory, yes? What's the intent behind _includeAddress?

Reply:
Not really, the factory doesn't create any metadata implementation (otherwise it would become too expensive to deploy).

`_includeAddress` adds the address of the exchange to the URI, otherwise it just uses the token ID, this is important because when using a single metadata implementation for all exchanges the URI needs to have information about the exchange address, not only the token id.